src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala

```diff
@@ -348,31 +440,32 @@ trait GraphDAO {
   def getTypeParams(tpe: Type) = tpe.asInstanceOf[TypeRefApi].args
 
   //SAVE METHODS
-  def saveObject[T <: DomainObject :TypeTag :ClassTag](obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
-    saveObject(typeOf[T], obj, vertex, wsc, graph)
+  def saveObject[T <: DomainObject :CachedType](obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
+    saveObject(cachedTypeOf[T], obj, vertex, wsc, graph)
   }
 
-  def saveObject(tpe: Type, obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
+  def saveObject(tpe: CachedType[_], obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Unit = {
     //Serialize out each of the case class properties.
     getPropertiesAndValues(tpe, obj).foreach({
       case (tp, prop, value) => saveProperty(tp, prop, value, vertex, wsc, graph)
     })
 
     if( wsc.isDefined ) {
-      saveProperty(typeOf[DateTime], "lastModified", DateTime.now, wsc.get.workspaceVertex, wsc, graph)
+      saveProperty(cachedTypeOf[DateTime], "lastModified", DateTime.now, wsc.get.workspaceVertex, wsc, graph)
     }
   }
 
   // a function to uniquely identify a graph object relative to siblings, by checking the idFields for equality
-  private def domainObjectFilterFn(tpe: Type, obj: DomainObject)(vertex: Vertex): Boolean = {
-    obj.idFields.forall { field => obj.getFieldValue(tpe, field) == vertex.getProperty(field) }
+  private def domainObjectFilterFn(tpe: CachedType[_], obj: DomainObject): Vertex => Boolean = {
+    val idFieldValues = obj.idFields.map(field => (field, obj.getFieldValue(tpe.tpe, field)))
+    (vertex: Vertex) => idFieldValues.forall { case (field, value) => value == vertex.getProperty(field) }
   }
 
-  def saveSubObject[T <: DomainObject :TypeTag :ClassTag](propName: String, obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Vertex = {
-    saveSubObject(typeOf[T], propName, obj, vertex, wsc, graph)
+  def saveSubObject[T <: DomainObject :TypeTag :ClassTag :CachedType](propName: String, obj: T, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Vertex = {
+    saveSubObject(cachedTypeOf[T], propName, obj, vertex, wsc, graph)
   }
 
-  def saveSubObject(tpe: Type, propName: String, obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Vertex = {
+  def saveSubObject(tpe: CachedType[_], propName: String, obj: DomainObject, vertex: Vertex, wsc: Option[WorkspaceContext], graph: Graph): Vertex = {
```

Comment:
can we make a typedef for `CachedType[_]` so i don't have to see that offensive `_` all over the code?